src/spec.c

```diff
@@ -418,23 +418,28 @@ static int IndexSpec_AddFieldsInternal(IndexSpec *sp, ArgsCursor *ac, QueryError
   FieldSpec *fs = NULL;
 
   while (!AC_IsAtEnd(ac)) {
-    size_t nfieldName = 0;
+    size_t nfieldName;
     const char *fieldName = AC_GetStringNC(ac, &nfieldName);
     if (IndexSpec_GetField(sp, fieldName, nfieldName)) {
-      QueryError_SetError(status, QUERY_EINVAL, "Duplicate field in schema");
+      QueryError_SetErrorFmt(status, QUERY_EINVAL, "Duplicate field in schema - %s", fieldName);
```

Comment:
Regarding security, is there any risk of overflow when passing the in QueryError_SetErrorFmt  fieldName ?

Reply:
@emmanuelkeller 
No. The function calls `rm_vasprintf` and creates another string that is later freed.